ddl/ddl_db_test.go

```diff
@@ -1560,6 +1560,16 @@ func (s *testDBSuite) TestCreateTableWithPartition(c *C) {
 	c.Assert(part.Definitions[2].Name, Equals, "p2")
 }
 
+func (s *testDBSuite) TestCreateTableWithFloatType(c *C) {
+	s.tk.MustExec("use test")
+	s.tk.MustExec("drop table if exists t")
+	s.testErrorCode(c, "create table t (a decimal(1, 2))", tmysql.ErrMBiggerThanD)
+	s.testErrorCode(c, "create table t (a float(1, 2))", tmysql.ErrMBiggerThanD)
+	s.testErrorCode(c, "create table t (a double(1, 2))", tmysql.ErrMBiggerThanD)
+	s.mustExec(c, "create table t (a double(1, 1))")
```

Comment:
Could we fix the case of "alter table t add column x decimal(1, 2);" ？

Reply:
Done, Good advise, we also check it in `change` and `modify` too~